staging/src/github.com/openshift/openshift-controller-manager/pkg/build/controller/build/build_controller.go

```diff
@@ -360,10 +373,57 @@ func (bc *BuildController) defaults() builddefaults.BuildDefaults {
 	return copy
 }
 
-func (bc *BuildController) setBuildDefaultProxy(spec *configv1.ProxySpec) {
+func (bc *BuildController) setBuildProxy(buildCfg *configv1.Build, globalProxy *configv1.Proxy) {
 	bc.configLock.Lock()
 	defer bc.configLock.Unlock()
-	bc.buildDefaults.DefaultProxy = spec
+
+	if buildCfg == nil && globalProxy == nil {
+		bc.buildDefaults.Config = nil
+		bc.buildDefaults.DefaultProxy = nil
+		return
+	}
+
+	cfg := bc.buildDefaults.Config
+	if cfg == nil {
+		cfg = &openshiftcontrolplanev1.BuildDefaultsConfig{}
+		bc.buildDefaults.Config = cfg
+	}
```

Comment:
Block here probably needs to be higher in the logic (so you can empty out the git proxy if `nil`s are passed in)

Reply:
part of next push